Tribler/Test/Core/Modules/RestApi/test_my_channel_endpoints.py

```diff
@@ -220,6 +224,31 @@ def test_rss_feeds_endpoint_no_my_channel(self):
         return self.do_request('mychannel/rssfeeds', expected_code=404)
 
 
+class MyChannelCreateTorrentEndpoint(AbstractTestMyChannelEndpoints):
+
+    @deferred(timeout=10)
+    def test_create_torrent(self):
+        torrent_path = os.path.join(TESTS_DATA_DIR, 'video.avi.torrent')
+        expected_tdef = TorrentDef.load(torrent_path)
+
+        def verify_torrent(body):
+            response = json.loads(body)
+            data = base64.b64decode(response["torrent"])
+            tdef = TorrentDef.load_from_memory(data)
+            creation_date = tdef.get_creation_date()
+            expected_tdef.metainfo["creation date"] = creation_date
+            self.assertEqual(expected_tdef, tdef)
+
+        post_data = {
+            "files": [os.path.join(TESTS_DATA_DIR, 'video.avi')],
+            "tracker": "http://localhost/announce",
+            "name": "Test torrent",
+            "description": "Video of my cat"
+        }
+        self.should_check_equality = False
+        return self.do_request('mychannel/createtorrent', 200, None, 'GET', post_data).addCallback(verify_torrent)
```

Comment:
That is what self.assertEqual(expected_tdef, tdef) should be doing in the callback verify_torrent